### ISSUE TITLE
chimera gets stuck in a broken state when out of space - hard to recover from

### ISSUE THREAD

**Issue by blurymind** [ChimeraOS/chimeraos#1026]
I left steam downloading games the other day and while it did - i had no warnings or anything. The next day when I booted into chimera,
it went straight to gnome shell. Again no warnings nothing.

When i tried to run literaly anything - even a terminal - nothing happened. No errors, nothing. Nothing could be started. I chouldnt open a web browser, a terminal or go into game mode and ChimeraOS wasnt telling me why.

I did ctrl+alt+f2 to see what was going on and after running some commands I could see that it had used 99% of the available space. I had to rm -r some folders and downloaded files. When I got it down to 94% or less and restarted - everything started working again.

Why is this bad?
The OS doesnt communicate to the user or warn them when its running out of space.
It just stops working and still continues to not tell them whats wrong.
It requires linux technical knowledge to know how to analyze whats wrong and fix it - a lot of beginner users will try to open the gnome shell terminal to fix it and then realize they are stuck because they cant. If I didnt know about the recovery keyboard shortcut, i would have wasted time reinstalling the whole thing.
Alot of users coming to this OS are gamers and it is possibly their first experience with linux outside of steamos.

Suggestion:
- out of space warnings in gnome shell
- recovery cli boot mode that detects when there is no more space and offers the user some options to clear some 
- recovery grub option - Chimera doesnt event have a recovery boot mode that boots it into terminal with no ui
- if its another issue - display logs when it boots - make it stop showing the fancy boot logo and let it show the logs when errors are encountered

**Comment by alkazar** [comment]
I am not sure how you got into such a state. Steam should not be allowing you to install games when there is not enough space on the drive.

We are not Steam or Gnome developers so I don't think there is anything we can really do here other than perhaps to show a warning in the startup splash screen assuming that even works under low disk space conditions.

**Comment by blurymind** [comment]
yeah it just keeps going and puts ChimeraOs in a broken state after a reboot.

Have you ever tested this scenario yourself? What does it do for you?

One thing that the OS can do is detect when space is low and offer some hint or solution or even to run a script to get you out of the broken state.

You could add to the start script a check on how much space there is left on the root partition and if its 99% full display something. One could even use zenith to create a pop up dialog warning when you boot into gnome shell - and if you go further even put a button on it that clears the various cache folders? Like a mini bleachbit? 

I am honestly not sure what else to suggest there or sure if zenith itself will show up if that happens. Surely SteamOS has some mechanism that is failing in ChimeraOs?
